cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -852,6 +853,35 @@ func startServer(conf rootConfig) error {
 	return nil
 }
 
+func newSourcegraphClient(rootURL *url.URL, hostname string, batchSize int) *sourcegraphClient {
+
+	client := retryablehttp.NewClient()
+	client.Logger = debug
+
+	// Sourcegraph might return an error message in the body if StatusCode==500. The
+	// default behavior of the go-retryablehttp client is to drain the body and not
+	// to propagate the error. Hence, we call ErrorPropagatedRetryPolicy instead of
+	// DefaultRetryPolicy and augment the error with the response body if possible.
+	client.CheckRetry = func(ctx context.Context, resp *http.Response, err error) (bool, error) {
+		shouldRetry, checkErr := retryablehttp.ErrorPropagatedRetryPolicy(ctx, resp, err)
+
+		if resp.StatusCode == http.StatusInternalServerError {
+			if b, e := io.ReadAll(resp.Body); e == nil {
+				checkErr = fmt.Errorf("%w: body=%q", checkErr, string(b))
```

Comment:
should we just be returning false for shouldRetry here?

Reply:
Not sure about this one. Just because Sourcegraph returns a 500 with an embedded error does not mean it is not recoverable. In general we don't know. I think retrying with backoff is still a good strategy.